### PR TITLE
"Recent status changes" Dashboard

### PR DESCRIPTION
**Pull request by georgehrke**
| Empty content | Statuses |
|:---------:|:------:|
|0A1E5196-00D4-4C8B-BA11-59062916FC62|7C962DF0-911E-4F10-845E-BB814AB295C9|

### CODE REVIEW

**Review by jancborchardt — CHANGES_REQUESTED** [review]
Super nice! :) Design feedback:
- [x] The icon should be black, we changed that for Files as well
- [x] The title is a bit long, let’s cut it to "Recent statuses"?
- [x] Should we add time info to the subline with a comma? Like "🌴 Vacationing, 3 days ago"

Edited by @georgehrke: Turned into checkboxes

File: apps/user_status/src/views/Dashboard.vue [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+<template>
+	<DashboardWidget
+		id="user-status_panel"
+		:items="statuses"
+		:loading="loading">
+		<template v-slot:default="{ item }">
+			<DashboardWidgetItem :item="getWidgetItem(item)">
+				{{ item }}
+			</DashboardWidgetItem>
+		</template>
```
Comment (eneiluj) [inline apps/user_status/src/views/Dashboard.vue:31]: You don't need to redefine the default slot for `DashboardWidget` component. It already uses `DashboardWidgetItem`.

```suggestion
	<DashboardWidget
		id="user-status_panel"
		:items="items"
		:loading="loading">
```
with `items` being a computed prop like that:
```
items() {
    return this.statuses.map((s) => {
        return this.getWidgetItem(s)
    })
},
```

**Review by julien-nc — CHANGES_REQUESTED** [review]
Very nice!

File: apps/user_status/src/views/Dashboard.vue (hunk 1 above)
Comment (georgehrke) [reply apps/user_status/src/views/Dashboard.vue]: Excellent point!

**Comment by georgehrke** [comment]
@juliushaertl @eneiluj Would you consider the misaligned empty content message an issue in nc/vue-dashboard?
(See original post)

**Comment by georgehrke** [comment]
Rebased and squashed

**Review by rullzer — APPROVED** [review]

File: apps/user_status/src/views/Dashboard.vue [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  -->
+
+<template>
+	<DashboardWidget
+		id="user-status_panel"
+		:items="items"
+		:loading="loading">
+		<template v-slot:empty-content>
+			<EmptyContent icon="icon-user-status">
+				{{ t('user_status', 'No recent status changes.') }}
```
Comment (jancborchardt) [inline apps/user_status/src/views/Dashboard.vue:29]: We generally don’t use dots at the end of these headings. :)

Comment (georgehrke) [reply apps/user_status/src/views/Dashboard.vue:29]: https://github.com/nextcloud/server/pull/22321 will cause merge conflicts, so let's get #22321 in first, then i will rebase and fix the wording

**Comment by georgehrke** [comment]
Rebased, squashed and fixed @jancborchardt's remark about the dot.

**Comment by rullzer** [comment]
Needs a rebase. But then good to go I think

**Comment by georgehrke** [comment]
Rebased again

**Review by skjnldsv — APPROVED** [review]

**Pull request merged**